detect/detect.go

```diff
@@ -413,3 +408,81 @@ func (d *Detector) addFinding(finding report.Finding) {
 func (d *Detector) addCommit(commit string) {
 	d.commitMap[commit] = true
 }
+
+// this function knows how to find the full line based on the secret and the newline chars it is between
+func findFullLine(fragment Fragment, loc Location, secret string) string {
+
+	secretLocation := findRelevantOccurenceOfSecret(fragment.Raw, secret, loc)
+	if secretLocation == nil {
+		return ""
+	}
+
+	secretStartingIdx := secretLocation[0] // start of secret in fragment
+	secretEndingIdx := secretLocation[1]   // end of secret in fragment
+
+	// find the nearest previous newline
+	prevNewlineIndex := findIndexAfterPreviousNewline(fragment.Raw, secretStartingIdx-1)
+	// find the nearest next newline
+	nextNewlineIndex := findIndexBeforeNextNewline(fragment.Raw, secretEndingIdx+1)
+
```

Comment:
again, protection vs minified files, must do a check that  we're not capturing more than 250 chars